lib/dev/kit/system.rb

```diff
@@ -124,14 +124,17 @@ def system(*a, sudo: false, env: ENV, **kwargs)
               err_r => ->(data) { STDOUT.write(data) }, }
           end
 
+          previous_trailing = Hash.new('')
           loop do
             ios = [err_r, out_r].reject(&:closed?)
             break if ios.empty?
 
             readers, = IO.select(ios)
             readers.each do |io|
               begin
-                handlers[io].call(io.readpartial(4096))
+                data, trailing = split_partial_characters(io.readpartial(4096))
```

Comment:
I bet we can call `#read` instead and pass it `encoding: Encoding::UTF_8` -- it looks like if we pass the right encoding (or if it has selected UTF-8 already) and don't pass a length, it should handle multibyte characters correctly.

Reply:
`read` without a length reads to EOF.  This bug came up in the context of a very long running command which gave output about what it was doing.  I don't think that's a good option.  `readline` might be good.  I'll try it.

Reply:
Readline may buffer until it gets a newline, which won't work well for spinners and the like. There has to be *something* we can do though; there are a whole lot of IO methods.

Reply:
Just played with it a bit, it looks like maybe there isn't an IO method that does what we want. Hmm.

Reply:
`readline` seemed to work well for the case this came up in.  I think the reality is that we want two behaviour modes.  For `block_given?`, we want `readline`, and `else` we want to spit out anything we get ASAP: `readpartial`.  I'll try reworking this for that.